scenes/game/local_game.gd

```diff
@@ -2793,6 +2795,9 @@ func is_effect_condition_met(performing_player : Player, effect, local_condition
 				if sealed_card.definition["id"] == card_id:
 					return false
 			return true
+		elif effect['condition_amount'] == "OPPONENT_SPEED":
```

Comment:
this is wrong.
the conditions being compared here are stuff in the "condition" field.
Then inside the elif block you should read additional fields like condition_amount.